src/sonic-config-engine/minigraph.py

```diff
@@ -413,6 +428,81 @@ def parse_deviceinfo(meta, hwsku):
                 port_speeds[port_alias_map.get(alias, alias)] = speed
     return port_speeds, port_descriptions
 
+# Special parsing for spine chassis frontend 
+def parse_spine_chassis_fe(results, vni, lo_intfs, phyport_intfs, pc_intfs, pc_members, devices):
+    chassis_vnet ='VnetFE'
+    chassis_vxlan_tunnel = 'TunnelInt'
+    chassis_vni = vni
+
+    # Vxlan tunnel information
+    lo_addr = '0.0.0.0'
+    for lo in lo_intfs:
+        lo_network = ipaddress.IPNetwork(lo[1])
+        if lo_network.version == 4:
+            lo_addr = str(lo_network.ip)
+            break        
+
+    results['VXLAN_TUNNEL'] = {chassis_vxlan_tunnel: {
+        'source_ip': lo_addr
+    }}
+
+    # Vnet information
+    results['VNET'] = {chassis_vnet: {
+        'vxlan_tunnel': chassis_vxlan_tunnel,
+        'vni': chassis_vni
+    }}
+
+    # Find physical L3 interfaces that should be enslaved to Vnet
+    for intf in phyport_intfs:
+        if isinstance(intf, tuple) == False:
```

Comment:
>isinstance [](start = 11, length = 10)

use this function.

    def is_ip_prefix_in_key(self, key):
        '''
        Function to check if IP address is present in the key. If it
        is present, then the key would be a tuple or else, it shall be
        be string
        '''
        return (isinstance(key, tuple))

Reply:
for better code readability, we need to directly understand the intention of the code without guessing.

Reply:
@lguohan  Fixed